silo: Fix HDF5 1.13 API breakage

chuckatkins opened the pull request.

Commit:
silo: Fix HDF5 1.13 API breakage

sethrj commented on var/spack/repos/builtin/packages/silo/package.py:
```diff
@@ -68,8 +68,10 @@ class Silo(AutotoolsPackage):
-    conflicts('hdf5@1.13:', when='+hdf5')
+    # API changes in hdf5-1.13 cause breakage
+    # See https://github.com/LLNL/Silo/pull/260
+    patch('hdf5-113.patch', when='@4.11:')
```
Make this conditional on the newer hdf5? Better not to patch code if we can help it

sethrj replied:
(To clarify: if the patch is unconditional, it'll invalidate every current installation of silo.)

```suggestion
    patch('hdf5-113.patch', when='@4.11: ^hdf5@1.13:')
```

sethrj reviewed (dismissed):
Thanks!

chuckatkins commented:
The updated patch just now removed some unrelated white space changes I accidentally let creep in.  I spoke with @brtnfld from The HDF Group and confirmed that this is a valid approach and works to address the API compatibility issues.  (We spoke directly but he also added his approval over on the upstream PR in https://github.com/LLNL/Silo/pull/260)
